### PR TITLE
build/configs/rtl8720e/: 1st commit for TP1E amebalite Tizenrt porting.

### PR DESCRIPTION
**Pull request by fengyunlai**
1st commit for TP1E amebalite Tizenrt porting.

### CODE REVIEW

File: build/configs/rtl8720e/hello/Make.defs [hunk 1]
```diff
@@ -0,0 +1,228 @@
+###########################################################################
```
Comment (sangam-swami) [inline build/configs/rtl8720e/hello/Make.defs:1]: Can you please create multiple commits with different changes related to arch, amebad, kernel etc..
It will be easy to review and modular

Comment (fengyunlai) [reply build/configs/rtl8720e/hello/Make.defs:1]: Hi Sangam, thank you for your advice. But it might be hard to separate the commits. In my personal developing branch, my colleagues and I have committed I think around 50 commits these days. It will be hard for me to summarize all commits and   reform them to different changes as suggested.

Comment (sunghan-chang) [reply build/configs/rtl8720e/hello/Make.defs:1]: How about doing as @sangam-swami says when we merge new chipset to master with porting completion. How do you think, @sangam-swami ?

Comment (sunghan-chang) [reply build/configs/rtl8720e/hello/Make.defs:1]: I agree @sangam-swami 's opinion.
Huge single commit is impossible to review.
I think @fengyunlai could do easily for @sangam-swami 's request.
```
git add apps/
git commit -s 
(1st commit for app)
git add build/
git commit -s
(2nd commit for build)
git add os/arch/
git add os/board/
git commit -s
(3rd commit for bsp)
...
```

Comment (fengyunlai) [reply build/configs/rtl8720e/hello/Make.defs:1]: Hi Mr Chang, yes, I can separate them by folders as you suggested.

Comment (sunghan-chang) [reply build/configs/rtl8720e/hello/Make.defs:1]: It's ok because the target branch is not the master. When we merge it into the master, debug log should be cleaned up like removal or converting to xxdbg.

Comment (fengyunlai) [reply build/configs/rtl8720e/hello/Make.defs:1]: Hi Mr Chang and Sangam, I just created 5 commits which is separated by folders, thanks

File: os/arch/arm/include/amebalite/amebalite_nsc.h [hunk 2]
```diff
@@ -0,0 +1,226 @@
+/*
```
Comment (thapav) [inline os/arch/arm/include/amebalite/amebalite_nsc.h:1]: Please add Samsung license as well

File: os/arch/arm/include/amebalite/chip.h [hunk 3]
```diff
@@ -0,0 +1,142 @@
+/****************************************************************************
+ *
+ * Copyright 2020 Samsung Electronics All Rights Reserved.
```
Comment (thapav) [inline os/arch/arm/include/amebalite/chip.h:3]: Change year to 2022 for this and other files

File: os/arch/arm/include/amebalite/amebalite_nsc.h (hunk 2 above)
Comment (fengyunlai) [reply os/arch/arm/include/amebalite/amebalite_nsc.h:18]: I have added as suggested, thank you

File: os/arch/arm/include/amebalite/chip.h (hunk 3 above)
Comment (fengyunlai) [reply os/arch/arm/include/amebalite/chip.h]: I have change all license year to 2022

File: build/configs/rtl8720e/board_metadata.txt [hunk 4]
```diff
@@ -0,0 +1,28 @@
+###########################################################################
+#
+# Copyright 2021 Samsung Electronics All Rights Reserved.
```
Comment (jeongarmy) [inline build/configs/rtl8720e/board_metadata.txt:3]: Please update the year, 2021 -> 2022.

File: build/configs/rtl8720e/hello/Make.defs [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+#
+# Copyright 2020 Samsung Electronics All Rights Reserved.
```
Comment (jeongarmy) [inline build/configs/rtl8720e/hello/Make.defs:3]: Please check and update the year to 2022.

File: build/configs/rtl8720e/board_metadata.txt (hunk 4 above)
Comment (fengyunlai) [reply build/configs/rtl8720e/board_metadata.txt]: Thank you! I have checked and update the rest files

File: build/configs/rtl8720e/hello/Make.defs (hunk 5 above)
Comment (fengyunlai) [reply build/configs/rtl8720e/hello/Make.defs]: Thank you! I have checked and update the rest files

File: os/arch/arm/src/Makefile [hunk 6]
```diff
@@ -1,6 +1,6 @@
 ###########################################################################
 #
-# Copyright 2016 Samsung Electronics All Rights Reserved.
+# Copyright 2022 Samsung Electronics All Rights Reserved.
```
Comment (sunghan-chang) [inline os/arch/arm/src/Makefile:3]: Please revert this change. For this file, previous is correct.

File: os/LibTargets.mk [hunk 7]
```diff
@@ -1,341 +1,357 @@
...
-$(LIBRARIES_DIR)$(DELIM)libwpa2$(LIBEXT): $(EXTDIR)$(DELIM)esp_idf_port/esp32$(DELIM)lib$(DELIM)libwpa2$(LIBEXT)
-	$(Q) install $(EXTDIR)$(DELIM)esp_idf_port/esp32$(DELIM)lib$(DELIM)libwpa2$(LIBEXT) $(LIBRARIES_DIR)$(DELIM)libwpa2$(LIBEXT)
-
-$(LIBRARIES_DIR)$(DELIM)libwpa$(LIBEXT): $(EXTDIR)$(DELIM)esp_idf_port/esp32$(DELIM)lib$(DELIM)libwpa$(LIBEXT)
-	$(Q) install $(EXTDIR)$(DELIM)esp_idf_port/esp32$(DELIM)lib$(DELIM)libwpa$(LIBEXT) $(LIBRARIES_DIR)$(DELIM)libwpa$(LIBEXT)
-
-$(LIBRARIES_DIR)$(DELIM)libwps$(LIBEXT): $(EXTDIR)$(DELIM)esp_idf_port/esp32$(DELIM)lib$(DELIM)libwps$(LIBEXT)
-	$(Q) install $(EXTDIR)$(DELIM)esp_idf_port/esp32$(DELIM)lib$(DELIM)libwps$(LIBEXT) $(LIBRARIES_DIR)$(DELIM)libwps$(LIBEXT)
-endif
+###########################################################################
```
Comment (sunghan-chang) [inline os/LibTargets.mk:1]: Please check this file change. I think you modify this file with Windows.

File: os/FlatLibs.mk [hunk 8]
```diff
@@ -1,197 +1,202 @@
...
-TINYARALIBS += $(LIBRARIES_DIR)$(DELIM)libwpa2$(LIBEXT)
-TINYARALIBS += $(LIBRARIES_DIR)$(DELIM)libespnow$(LIBEXT)
-TINYARALIBS += $(LIBRARIES_DIR)$(DELIM)libphy$(LIBEXT)
-TINYARALIBS += $(LIBRARIES_DIR)$(DELIM)libmesh$(LIBEXT)
-endif
-
-# Export all libraries
-EXPORTLIBS = $(TINYARALIBS)
-
+###########################################################################
```
Comment (sunghan-chang) [inline os/FlatLibs.mk:1]: same as previous

File: framework/src/wifi_manager/wifi_manager_state.c [hunk 9]
```diff
@@ -322,32 +322,34 @@ wifi_manager_result_e _wifimgr_scan(wifi_manager_scan_config_s *config)
 
 wifi_manager_result_e _handler_on_uninitialized_state(wifimgr_msg_s *msg)
 {
-	wifimgr_evt_e evt = msg->event;
-	if (evt != WIFIMGR_CMD_INIT) {
-		WIFIADD_ERR_RECORD(ERR_WIFIMGR_INVALID_EVENT);
-		return WIFI_MANAGER_FAIL;
-	}
+ wifimgr_evt_e evt = msg->event;
```
Comment (sunghan-chang) [inline framework/src/wifi_manager/wifi_manager_state.c:325]: This looks something wrong. Could you check?

File: external/dhcpc/dhcpc_lwip.c [hunk 10]
```diff
@@ -94,7 +94,6 @@ int dhcp_client_start(const char *intf)
 {
 	int ret = -1;
 	struct req_lwip_data req;
-
```
Comment (sunghan-chang) [inline external/dhcpc/dhcpc_lwip.c:97]: Let's revert unnecessary changes in this file.

File: apps/examples/testcase/ta_tc/systemio/itc/itc_gpio.c [hunk 11]
```diff
@@ -25,7 +25,7 @@
 #include <iotbus_gpio.h>
 #include <iotbus_error.h>
 
-#define GPIO_PIN 41
+#define GPIO_PIN 25
```
Comment (sunghan-chang) [inline apps/examples/testcase/ta_tc/systemio/itc/itc_gpio.c:28]: No need to modify this app change for tc because this will be merged into the RTL8720E_support branch, not the master. But tc should be a common module regardless of the board so that we will modify this logic later. So, when we merge RTL8720E code into the master eventually, we should not add this change. Just FYI.

File: os/board/rtl8720e/src/Makefile [hunk 12]
```diff
@@ -0,0 +1,205 @@
...
+CFLAGS += -I$(COMPONENT_DIR)/wifi/inic
+CFLAGS += -I$(COMPONENT_DIR)/wifi/wifi_config
+CFLAGS += -I$(COMPONENT_DIR)/wifi/wifi_fw/amebalite/include
+CFLAGS += -I$(COMPONENT_DIR)/wifi/driver/include
+CFLAGS += -I$(COMPONENT_DIR)/wifi/driver/src/hal/phydm
+CFLAGS += -I$(COMPONENT_DIR)/wifi/driver/src/hal/rtl8720a
+CFLAGS += -I$(COMPONENT_DIR)/wifi/driver/src/osdep
+CFLAGS += -I$(COMPONENT_DIR)/wifi/driver/src/hal
+CFLAGS += -I$(COMPONENT_DIR)/wifi/driver/src/hci
+CFLAGS += -I$(COMPONENT_DIR)/wifi/driver/src/phl
```
[the remark this answers is not in the document]
Comment (sunghan-chang) [reply os/board/rtl8720e/src/Makefile:137]: This could be done later.

File: apps/examples/testcase/ta_tc/systemio/itc/itc_gpio.c (hunk 11 above)
Comment (sangam-swami) [reply apps/examples/testcase/ta_tc/systemio/itc/itc_gpio.c:28]: yes we will make it generic change in tc and add board dependent values in config

File: os/net/lwip/src/netif/Make.defs [hunk 13]
```diff
@@ -22,7 +22,9 @@
 ############################################################################
 
 
-LWIP_CSRCS += ethernet.c slipif.c
+LWIP_CSRCS += ethernet.c \
+			slipif.c \
+			ethernetif.c
```
[the remark this answers is not in the document]
Comment (fengyunlai) [reply os/net/lwip/src/netif/Make.defs:27]: Dear Mr Chang, I have checked internally with the owner of that related commit, we can and I will revert this. Sorry for confusion.

File: os/arch/arm/src/Makefile (hunk 6 above)
Comment (fengyunlai) [reply os/arch/arm/src/Makefile:3]: Yes Mr Chang I will revert to 2016.

File: framework/src/wifi_manager/wifi_manager_state.c (hunk 9 above)
Comment (fengyunlai) [reply framework/src/wifi_manager/wifi_manager_state.c:325]: Yes, we will revert this as well.

File: external/dhcpc/dhcpc_lwip.c (hunk 10 above)
Comment (fengyunlai) [reply external/dhcpc/dhcpc_lwip.c:97]: Yes sure.

File: apps/examples/testcase/ta_tc/systemio/itc/itc_gpio.c (hunk 11 above)
Comment (fengyunlai) [reply apps/examples/testcase/ta_tc/systemio/itc/itc_gpio.c:28]: Thank you Mr Chang and Sangam, then we will revert all tc temp changes, and will add related defconfig Macro after the changes you mentioned.

File: os/dbuild.sh [hunk 14]
```diff
@@ -84,6 +84,8 @@ function FIND_BINFILE()
 			BINFILE="${BINDIR}/tinyara_master_strip"
 		elif [[ "${CONFIG_ARCH_BOARD}" == "rtl8721csm" ]]; then
 			BINFILE="${BINDIR}/${KERNEL_BIN_NAME}"
+		elif [[ "${CONFIG_ARCH_BOARD}" == "rtl8720e" ]]; then
+			BINFILE="${BINDIR}/${KERNEL_BIN_NAME}"
```
Comment (sangam-swami) [inline os/dbuild.sh:88]: BINFILE name is same, you can put in || condition with rtl8721csm

Comment (sunghan-chang) [reply os/dbuild.sh:88]: We are modifying this function. Let's merge it as it is.

File: os/board/rtl8720e/src/Makefile (hunk 12 above)
[the remark this answers is not in the document]
Comment (fengyunlai) [reply os/board/rtl8720e/src/Makefile:137]: I will take note and our team will clean up later in other commits

File: os/LibTargets.mk (hunk 7 above)
Comment (fengyunlai) [reply os/LibTargets.mk]: Change back to directly edit with Linux

File: os/FlatLibs.mk (hunk 8 above)
Comment (fengyunlai) [reply os/FlatLibs.mk]: Change back to directly edit with Linux

File: build/configs/rtl8720e/hello/Make.defs [hunk 15]
```diff
@@ -0,0 +1,228 @@
...
+
+ifneq ($(CONFIG_DEBUG_NOOPT),y)
+  ARCHOPTIMIZATION += $(MAXOPTIMIZATION) -fno-strict-aliasing -fno-strength-reduce -fomit-frame-pointer
+endif
+
+ifeq ($(CONFIG_FRAME_POINTER),y)
+  ARCHOPTIMIZATION += -fomit-frame-pointer -mapcs -mno-sched-prolog
+endif
+
+ARCHCFLAGS = -fno-builtin -mcpu=cortex-m33 -mfpu=fpv5-sp-d16
```
[the remark this answers is not in the document]
Comment (fengyunlai) [reply build/configs/rtl8720e/hello/Make.defs:123]: Thank you~ But maybe I think we'd better change it at the same time when "-march=armv8.0-m.main+dsp" is change to "-march=armv8.1-m.main+dsp". What's your opinion?

File: os/KernelLibs.mk [hunk 16]
```diff
@@ -143,6 +143,11 @@ TINYARALIBS += $(LIBRARIES_DIR)$(DELIM)lib_wlan$(LIBEXT)
 TINYARALIBS += $(LIBRARIES_DIR)$(DELIM)lib_wps$(LIBEXT)
 endif
 
+ifeq ($(CONFIG_AMEBALITE_WIFI),y)
+TINYARALIBS += $(LIBRARIES_DIR)$(DELIM)lib_enc$(LIBEXT)
```
Comment (sunghan-chang) [inline os/KernelLibs.mk:147]: 1. Could you share what the lib_enc is?
2. Do we need this with commented out?
3. Only this is not commented. Could you check?
4. Please add them in the protectedlibs.mk.

File: build/configs/rtl8720e/board_metadata.txt [hunk 17]
```diff
@@ -0,0 +1,28 @@
...
+###########################################################################
+# board_metadata.txt
+
+BOARD="RTL8720E"
+
+DEFAULT_PORT="ttyUSB1"
+
+BL1=""
+BL2=""
+KERNEL=""
```
Comment (sunghan-chang) [inline build/configs/rtl8720e/board_metadata.txt:26]: Could you add the output name like km0_km4_image2.bin?

File: os/KernelLibs.mk (hunk 16 above)
Comment (fengyunlai) [reply os/KernelLibs.mk:147]: Dear Mr Chang, lib_enc stands for encryption lib, it includes some algorithms for wifi.
In current stage, we do not need it. But in future commit, let's say next month, especially for wifi performance tuning, we might include it.

So I suggest we keep them commented out first.

And I will add them in protectedlibs.mk.

File: build/configs/rtl8720e/board_metadata.txt (hunk 17 above)
Comment (fengyunlai) [reply build/configs/rtl8720e/board_metadata.txt:26]: The output for 8720e is actually kr4_km4_image2.bin, and yes, I will add it and others like 8721csm

File: build/configs/rtl8720e/hello/Make.defs (hunk 15 above)
[the remark this answers is not in the document]
Comment (sangam-swami) [reply build/configs/rtl8720e/hello/Make.defs:123]: ok fine,

File: build/configs/rtl8720e/tc/Make.defs [hunk 18]
```diff
@@ -0,0 +1,228 @@
...
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+#
+###########################################################################
+############################################################################
+#
```
Comment (amogh-samsung) [inline build/configs/rtl8720e/tc/Make.defs:19]: Do we need this license since this is a new file?

File: os/arch/arm/include/amebalite/chip.h [hunk 19]
```diff
@@ -0,0 +1,142 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ *
```
Comment (amogh-samsung) [inline os/arch/arm/include/amebalite/chip.h:19]: You have added the Nuttx licenses at a few places.
Are all these files picked up from Nuttx for reference?

File: os/arch/arm/include/amebalite/irq.h [hunk 20]
```diff
@@ -0,0 +1,160 @@
...
+#define AMEBALITE_IRQ_DBGMONITOR    (12) /* Vector 12: Debug Monitor */
+                                     /* Vector 13: Reserved */
+#define AMEBALITE_IRQ_PENDSV        (14) /* Vector 14: Pendable system service request */
+#define AMEBALITE_IRQ_SYSTICK       (15) /* Vector 15: System tick */
+
+/* External interrupts (vectors >= 16).  These definitions are chip-specific */
+
+#define AMEBALITE_IRQ_FIRST         (16) /* Vector number of the first external interrupt */
+
+#define AMEBALITE_IRQ_WIFI_FISR_FESR				(AMEBALITE_IRQ_FIRST + 0)
```
Comment (amogh-samsung) [inline os/arch/arm/include/amebalite/irq.h:86]: Could  you modify the tab spaces so that all Macro definitions have the same indentation please?

Comment (sunghan-chang) [reply os/arch/arm/include/amebalite/irq.h:86]: @amogh-samsung We don't guide to follow TizenRT coding rule for chipset codes like arch, board and so on. How about keeping them as the chipset vendor does?

**Review by sangam-swami — APPROVED** [review]
Looks good

File: os/arch/arm/include/amebalite/irq.h (hunk 20 above)
Comment (sangam-swami) [reply os/arch/arm/include/amebalite/irq.h:86]: yes we agreed to give exception for vendor code, let vendor follow their standard.

**Review by sunghan-chang — APPROVED** [review]
